File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/series2d/HeatmapSeries2D.java
package com.scichart.docsandbox.examples.java.series2d;

import static com.scichart.drawing.utility.ColorUtil.Chartreuse;
import static com.scichart.drawing.utility.ColorUtil.CornflowerBlue;
import static com.scichart.drawing.utility.ColorUtil.DarkBlue;
import static com.scichart.drawing.utility.ColorUtil.DarkGreen;
import static com.scichart.drawing.utility.ColorUtil.Red;
import static com.scichart.drawing.utility.ColorUtil.Yellow;

import android.view.LayoutInflater;

import androidx.annotation.NonNull;

import com.scichart.charting.model.dataSeries.UniformHeatmapDataSeries;
import com.scichart.charting.modifiers.CursorModifier;
import com.scichart.charting.modifiers.ModifierGroup;
import com.scichart.charting.visuals.SciChartHeatmapColourMap;
import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.axes.NumericAxis;
import com.scichart.charting.visuals.renderableSeries.ColorMap;
import com.scichart.charting.visuals.renderableSeries.FastUniformHeatmapRenderableSeries;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.databinding.ExampleHeatmapChartFragmentBinding;
import com.scichart.docsandbox.examples.base.ExampleBaseFragment;

import java.util.Collections;

@ExampleDefinition()
public class HeatmapSeries2D extends ExampleBaseFragment<ExampleHeatmapChartFragmentBinding> {
    private static final int WIDTH = 200, HEIGHT = 300;

    @NonNull
    @Override
    protected ExampleHeatmapChartFragmentBinding inflateBinding(@NonNull LayoutInflater inflater) {
        return ExampleHeatmapChartFragmentBinding.inflate(inflater);
    }

    protected void initExample(@NonNull ExampleHeatmapChartFragmentBinding binding) {
        initExample(binding.surface);
    }

    public ModifierGroup createDefaultModifiers() {
        return sciChartBuilder.newModifierGroupWithDefaultModifiers().build();
    }

    // <Example>
    protected void initExample(@NonNull SciChartSurface surface) {
        final NumericAxis xAxis = new NumericAxis(requireContext());
        final NumericAxis yAxis = new NumericAxis(requireContext());

        final UniformHeatmapDataSeries<Integer, Integer, Double> dataSeries = new UniformHeatmapDataSeries<>(Integer.class, Integer.class, Double.class, WIDTH, HEIGHT);

        final FastUniformHeatmapRenderableSeries heatmapRenderableSeries = new FastUniformHeatmapRenderableSeries();
        heatmapRenderableSeries.setColorMap(
                new ColorMap(
                        new int[]{DarkBlue, CornflowerBlue, DarkGreen, Chartreuse, Yellow, Red},
                        new float[]{0f, 0.2f, 0.4f, 0.6f, 0.8f, 1}
                )
        );

        heatmapRenderableSeries.setMinimum(0);
        heatmapRenderableSeries.setMaximum(200);
        heatmapRenderableSeries.setDataSeries(dataSeries);

        final SciChartHeatmapColourMap colourMap = binding.heatmapColourMap;
        colourMap.setMinimum(heatmapRenderableSeries.getMinimum());
        colourMap.setMaximum(heatmapRenderableSeries.getMaximum());
        colourMap.setColorMap(heatmapRenderableSeries.getColorMap());

        Collections.addAll(surface.getXAxes(), xAxis);
        Collections.addAll(surface.getYAxes(), yAxis);
        Collections.addAll(surface.getRenderableSeries(), heatmapRenderableSeries);

        final CursorModifier cursor = new CursorModifier();
        cursor.setShowTooltip(true);
        cursor.setReceiveHandledEvents(true);

        ModifierGroup modifiers = createDefaultModifiers();
        modifiers.getChildModifiers().add(cursor);

        Collections.addAll(surface.getChartModifiers(), modifiers);
    }
    // </Example>
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/series2d/HeatmapSeries2D.kt
package com.scichart.docsandbox.examples.kotlin.series2d

import android.view.LayoutInflater
import com.scichart.charting.model.dataSeries.UniformHeatmapDataSeries
import com.scichart.charting.modifiers.CursorModifier
import com.scichart.charting.modifiers.ModifierGroup
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.axes.NumericAxis
import com.scichart.charting.visuals.renderableSeries.ColorMap
import com.scichart.charting.visuals.renderableSeries.FastUniformHeatmapRenderableSeries
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.databinding.ExampleHeatmapChartFragmentBinding
import com.scichart.docsandbox.examples.base.ExampleBaseFragment
import com.scichart.drawing.utility.ColorUtil
import java.util.*

@ExampleDefinition()
class HeatmapSeries2D : ExampleBaseFragment<ExampleHeatmapChartFragmentBinding>() {
    private val WIDTH = 200
    private val HEIGHT = 300

    override fun inflateBinding(inflater: LayoutInflater): ExampleHeatmapChartFragmentBinding {
        return ExampleHeatmapChartFragmentBinding.inflate(inflater)
    }

    override fun initExample(binding: ExampleHeatmapChartFragmentBinding) {
        initExample(binding.surface)
    }

    fun createDefaultModifiers(): ModifierGroup {
        return sciChartBuilder.newModifierGroupWithDefaultModifiers().build()
    }

    // <Example>
    fun initExample(surface: SciChartSurface) {
        val xAxis = NumericAxis(requireContext())
        val yAxis = NumericAxis(requireContext())

        val dataSeries = UniformHeatmapDataSeries(Int::class.java, Int::class.java, Double::class.java, WIDTH, HEIGHT)

        val heatmapRenderableSeries = FastUniformHeatmapRenderableSeries()
        heatmapRenderableSeries.colorMap = ColorMap(
            intArrayOf(ColorUtil.DarkBlue, ColorUtil.CornflowerBlue, ColorUtil.DarkGreen, ColorUtil.Chartreuse, ColorUtil.Yellow, ColorUtil.Red),
            floatArrayOf(0f, 0.2f, 0.4f, 0.6f, 0.8f, 1f)
        )

        heatmapRenderableSeries.minimum = 0.0
        heatmapRenderableSeries.maximum = 200.0
        heatmapRenderableSeries.dataSeries = dataSeries

        val colourMap = binding.heatmapColourMap
        colourMap.minimum = heatmapRenderableSeries.minimum
        colourMap.maximum = heatmapRenderableSeries.maximum
        colourMap.colorMap = heatmapRenderableSeries.colorMap

        Collections.addAll(surface.xAxes, xAxis)
        Collections.addAll(surface.yAxes, yAxis)
        Collections.addAll(surface.renderableSeries, heatmapRenderableSeries)

        val cursor = CursorModifier()
        cursor.showTooltip = true
        cursor.receiveHandledEvents = true

        val modifiers = createDefaultModifiers()
        modifiers.childModifiers.add(cursor)

        Collections.addAll(surface.chartModifiers, modifiers)
    }
    // </Example>
}
